hack/jenkins/setup_docker_desktop_macos.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Copyright 2021 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -x
+
+# kill docker first
+osascript -e 'quit app "Docker"'
```

Comment:
why kill docker if it is already running ? this could cause containers who have a volumes attached to them get stuck and not be remove-able.

how about restart it only if it is not already running successfully ?

Reply:
I was just following what we normally do, which is to always start fresh with a completely new docker.

Reply:
it would be easy to add a check to see if docker is already running, but our tests shouldn't be leaving containers laying around either way

Reply:
done